src/components/atoms/UserAvatar/UserAvatar.tsx

```diff
@@ -37,7 +42,26 @@ export const UserAvatar: React.FC<UserAvatarProps> = ({
     "user-avatar--clickable": onClick !== undefined,
   });
 
+  const isOnline = recentWorldUsers.find(
+    (worldUser) => worldUser.id === user?.id
+  );
```

Comment:
I don't think it's the most efficient way to decide if a person is `online`, but I don't have any other suggestions. At least this should be wrapper into useMemo. @0xdevalias wdyt?

Reply:
https://github.com/sparkletown/sparkle/pull/1384/commits/dfe92ea3574c1de767c22b418ec2a106109a6390

Reply:
@mike-lvov `recentWorldUsers` would seem to align with other similar patterns that we have used elsewhere I believe?

Looking at where `UserAvatar` is used in existing code:

- `src/components/organisms/ChatSidebar/components/PrivateChatPreview/PrivateChatPreview.tsx`
  - passes in `isOnline` from props
- `src/components/organisms/ChatSidebar/components/PrivateChats/PrivateChats.tsx`
  - uses `onlineUsers.some((user) => user.id === chatMessage.counterPartyUser.id)`
  - which is set from `const onlineUsers = useOnlineUsersToDisplay();`
- `src/hooks/privateChats.ts`
  - which uses `useRecentWorldUsers`